server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java

```diff
@@ -390,4 +393,21 @@ private void serveWellKnown(ServletRequestDetails request) {
       ExceptionUtil.throwRuntimeExceptionAndLog(logger, e.getMessage(), e);
     }
   }
+
+  private void preprocessRequest(
+      ServletRequestDetails servletRequestDetails, AccessDecision accessDecision) {
+    RequestMutation mutation =
+        accessDecision.preprocess(new RequestDetailsToReader(servletRequestDetails));
+    if (mutation == null || CollectionUtils.isEmpty(mutation.getQueryParams())) {
+      return ;
+    }
+
+    mutation
+        .getQueryParams()
+        .forEach((key, value) -> servletRequestDetails.addParameter(
```

Comment:
In the interface documentation, please also document that when there is a conflict (i.e., a returned parameter in `RequestMutation` is already present in the original request) then the new value overwrites the old one.